src/participant.rs

```diff
@@ -285,38 +288,26 @@ pub(crate) trait InnerProtocolParticipant: ProtocolParticipant {
 
     /// Process a `ready` message: tell other participants that we're ready and
     /// see if all others have also reported that they are ready.
-    fn process_ready_message<T: TypeTag<Value = ()>>(
+    fn process_ready_message<R: RngCore + CryptoRng>(
         &mut self,
+        rng: &mut R,
         message: &Message,
     ) -> Result<(ProcessOutcome<Self::Output>, bool)> {
-        self.local_storage_mut().store::<T>(message.from(), ());
-        let empty: [u8; 0] = [];
-        // If message came from self, then tell the other participants that we are ready
-        let self_initiated_outcome = if message.from() == self.id() {
-            let messages = self
-                .other_ids()
+        // If message came from self, then we should start the protocol
+        if message.from() == self.id() {
```

Comment:
suggestion: This is an aesthetic preference, but I prefer to do an explicit early return for the error and then un-indent the rest of the function, like:
```rust
if message.from != self {
   error!(...)
   Err(ProtocolError)?
}

let banked_messages = self.fetch_all...
```

To me that makes it more obvious that there's no case where we accept a ready message from someone else and also generally reducing indenting is nice. Up to you though.